### PR TITLE
Replace Poetry Based Project Configuration with setuptools

### LINKED ISSUE #7: Set up publishing workflow to pypi
Let's introduce standard github workflow setup for python projects

- undo poetry, use pip + pyproject.toml
- run tests on PR to main
- create a Release on merge to main
- release workflow publishes package to pypi

### PR DESCRIPTION
**Pull request by dipanjank**
Fixes #7 (partly)

# Proposed Changes

- Replace poetry related configuration in pyproject.toml with [PEP 518 supported configuration elements](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html) understood by pip.
- Change the CI/CD workflow to "pip install" the dependencies. 
- Move the pre-commit check in CI/CD to its own job.
- Add a simple PR template.

# Maintenance

* [ ] Updated tests
* [ ] Updated documentation

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,41 +1,55 @@
...
+[project]
 name = "sagemaker-rightline"
+requires-python = ">=3.8"
 version = "0.0.0"
-description = ""
-authors = ["sist <15859432+stiebels@users.noreply.github.com>"]
-license = "MIT"
+description = "Python package to easily validate properties of a SageMaker Pipeline."
+authors = [
+    {name = "stiebels", email = "stiebels@github.com"}
```
Comment (stiebels) [inline pyproject.toml:7]: Feel free to add your alias here as well

**Review by stiebels — CHANGES_REQUESTED** [review]
Few minor change requests

File: pyproject.toml [hunk 2]
```diff
@@ -1,41 +1,55 @@
...
+    "build",
+]
 
-[tool.poetry.group.dev.dependencies]
-moto = {extras = ["all"], version = "^4.1.10"}
-pytest = "^7.3.1"
-pytest-cov = "^4.0.0"
-pyproject-flake8 = "^6.0.0.post1"
-jupyter = "^1.0.0"
+[project.urls]  # Optional
```
[the remark this answers is not in the document]
Comment (dipanjank) [reply pyproject.toml:45]: Fixed.

File: .github/workflows/python-package.yml [hunk 3]
```diff
@@ -33,23 +38,17 @@ jobs:
         python-version: ${{ matrix.python-version }}
     - name: Pre-Commit
```
[the remark this answers is not in the document]
Comment (dipanjank) [reply .github/workflows/python-package.yml]: Fixed.

File: pyproject.toml (hunk 1 above)
Comment (dipanjank) [reply pyproject.toml:7]: Done.

**Review by stiebels — APPROVED** [review]

**Pull request merged**